crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -590,12 +592,16 @@ impl LogletConfiguration {
 
         match logs_configuration.default_provider {
             #[cfg(any(test, feature = "memory-loglet"))]
-            ProviderConfiguration::InMemory => {
-                Some(LogletConfiguration::Memory(loglet_id.next().into()))
-            }
-            ProviderConfiguration::Local => {
-                Some(LogletConfiguration::Local(loglet_id.next().into()))
-            }
+            ProviderConfiguration::InMemory => Some(LogletConfiguration::Memory(
+                loglet_id
+                    .map(|id| id.into())
+                    .unwrap_or_else(rand::random::<u64>),
+            )),
+            ProviderConfiguration::Local => Some(LogletConfiguration::Local(
+                loglet_id
+                    .map(|id| id.into())
+                    .unwrap_or_else(rand::random::<u64>),
+            )),
```

Comment:
I kind of liked that in the normal case we would also use the properly formatted `LogletIds` for the local and in-memory loglets. Could we maybe say if the previous type was not replicated then we `LogletId::new(log_id, last_segment_index.next())` and if it was, then `self.loglet_id().next()`?

Reply:
But that is implied here, no? `loglet_id` is None if current segment is local/memory, otherwise it will hold the value of the current replicated `loglet-id`. The next replicated loglet will either do `next()` (coming from a previous replicated loglet) or create a new one based on the segment index if isn't.

Reply:
ah, I think the `into() in this case should also be next

Reply:
Yes, but now all in memory and local loglets (also those that are created with the 1.2) have random loglet ids.

Reply:
So here is the logic:

Local -> Local (uses random loglet id) # compatible with < v1.2
Local -> replicated (use LogletId::new(log_id, segment_index.next())
replicated -> replicated (loglet_id.next())
replicated -> local (although not supported, will still use loglet_id.next())

Reply:
@tillrohrmann can you check again please